File: JetpacReloaded/C# version/JetpacReloaded/Bullet.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace JetPacReloaded
{
    public class Bullet : BaseObject
    {
        private bool m_left = false;

        public Bullet(int x, int y, Texture2D image, bool left)
        {
            m_image = image;
            m_left = left;
            m_screenLocation = new Vector2(x, y);
        }

        public void Update()
        {
            if (m_screenLocation.X < 800 && !m_left)
            {
                m_screenLocation.X += 6;
            }
            else if (m_screenLocation.X > -40)
            {
                m_screenLocation.X -= 6;
            }
            if (m_screenLocation.X is >= 800 or <= -40)
            {
                Offscreen = true;
            }
        }

        public bool Offscreen { get; private set; }

        public Rectangle BulletRect => new((int)m_screenLocation.X, (int)m_screenLocation.Y, m_image.Width, m_image.Height);
    }
}

File: JetpacReloaded/Typescript version/WebGLCanvas/ts/bullet.ts
import { Rectangle } from "../ts_lib/rectangle.js";
import { Texture2D } from "../ts_lib/texture2D.js";
import { Vector2 } from "../ts_lib/vector2.js";
import { BaseObject } from "./baseobject.js";

export class Bullet extends BaseObject {
    private m_left: boolean = false;
    private m_offscreen: boolean = false;

    constructor(x: number, y: number, image: Texture2D, left: boolean) {
        super();
        this.m_image = image;
        this.m_left = left;
        this.m_screenLocation = new Vector2(x, y);
    }
    public Update(): void {
        if (this.m_screenLocation.X < 800 && !this.m_left) {
            this.m_screenLocation.X += 6;
        }
        else if (this.m_screenLocation.X > -40) {
            this.m_screenLocation.X -= 6;
        }
        if (this.m_screenLocation.X >= 800 || this.m_screenLocation.X <= -40) {
            this.m_offscreen = true;
        }
    }
    public get Offscreen(): boolean {
        return this.m_offscreen;
    }
    public get BulletRect(): Rectangle {
        return new Rectangle(this.m_screenLocation.X, this.m_screenLocation.Y, this.m_image.width, this.m_image.height);
    }
}
